packages/fetchai/connections/gym/connection.py

```diff
@@ -96,8 +95,10 @@ def handle_gym_message(self, envelope: Envelope) -> None:
         :param envelope: the envelope
         :return: None
         """
-        gym_message = GymSerializer().decode(envelope.message)
-        gym_message = cast(GymMessage, gym_message)
+        assert isinstance(
+            envelope.message, GymMessage
+        ), "Message not of type GymMessage"
+        gym_message = cast(GymMessage, envelope.message)
```

Comment:
Note, connections no longer have to encode/decode if they don't use envelopes for transport or are never leaving the process